### PR TITLE
[WB-4362] add more to test docs

### PR DESCRIPTION
**Pull request by raubitsj**
https://wandb.atlassian.net/browse/WB-4362

Description
-----------

- Adds section talking about test points
- updates code coverage section to be up-to-date with current checks

Testing
-------

It is a document, no testing

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -135,6 +135,46 @@ def test_one_cell(notebook):
...
+  | Up |  .  | Sq |  .  | Ip |   .  | Ms |
+  +----+  .  +----+  .  +----+   .  +----+
+    |     .    |     .    |      .    |
+    | ------>  | -------> | --------> |    1
+    |     .    |     .    |      .    |
+    |     .    | -------> | --------> |    2
+    |     .    |     .    |      .    |
+    | ------>  |     .    |      .    |    3
+    |     .    |     .    |      .    |
+
```
Comment (KyleGoyette) [inline CONTRIBUTING.md:161]: Big fan of these diagrams.

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -161,18 +201,16 @@ We use codecov to ensure we're executing all branches of logic in our tests.  Be
...
-
-I plan on adding more categories, as we get some of these tests and fixtures improved:
-
-1. patch/internal: should be covered very high, the trick is all the error handling cases which I am working on
-2. patch/api: we have no good fixtures for this, so until we do, this will get a waiver
-3. patch/sdk-other: will be a catch all for other stuff in wandb/sdk/,  so patch/other will be stuff outside sdk
+3. patch/tests-utils: tests/conftest.py and supporting fixtures at tests/utils/, no coverage requirements
+4. patch/sdk: anything that matches wandb/sdk/*.py (so top level sdk files).   These have lots of ways to test, so it should be high coverage.  currently target is ~80% (but it is dynamic)
+5. patch/sdk-internal: should be covered very high target is around 80% (also dynamic)
+6. patch/sdk-other: will be a catch all for other stuff in wandb/sdk/
```
Comment (KyleGoyette) [inline CONTRIBUTING.md:211]: Do we have a target for this one?

**Review by KyleGoyette — APPROVED** [review]
Approved, though I did have one question that could potentially be covered in the doc.

File: CONTRIBUTING.md (hunk 2 above)
Comment (raubitsj) [reply CONTRIBUTING.md]: It is around 75% right now.  added. thanks.

**Pull request merged**